client/pages/clubs/[id].tsx

```diff
@@ -1,22 +1,228 @@
+      </IconContainer>
+
+      <Logo src={iconURL} alt="" />
+      <ClubName>{name}</ClubName>
+
+      <Links>
+        <SocialLink>techplusuw.ca</SocialLink>
+        <SocialLink style={{ marginLeft: '10px' }}>@techplusuw</SocialLink>
+      </Links>
+
+      <ClubMetaData>
+        <MemberCount>20-30 members</MemberCount>
+
+        {/* TODO: color code the tags */}
+        <Categories>
+          {tags.map((t) => (
+            <Category>{t}</Category>
+          ))}
+        </Categories>
+      </ClubMetaData>
+
+      <ClubDescription>{description}</ClubDescription>
+    </div>
+  )
+}
+
+const EventsTitle = styled.h2``
+
+const EventCardDetails = styled.div`
+  margin: 0 20px 20px;
+  width: 100%;
+
+  @media ${device.mobileS} {
+    margin: 0 10px 10px;
+  }
+`
+
+const EventCardTitle = styled.p`
+  font-weight: 700;
+  margin-bottom: 0;
+`
+
+// TODO: make box shadow cleaner
+const EventCard = styled.div`
+  box-shadow: 0 5px 10px rgba(154, 160, 185, 0.05), 0 15px 100px rgba(166, 173, 201, 0.2);
+  display: flex;
+  margin-top: 10px;
+
+  @media ${device.mobileL} {
+    font-size: 10px;
+  }
+`
+
+const EventCardImg = styled.img`
+  width: 128px;
+  @media ${device.mobileL} {
+    width: 64px;
+  }
+`
+
+const EventCardDate = styled.p`
+  margin-top: 5px;
+`
+
+const PastEventsTitle = styled.p`
+  font-size: ${desktopFontSize.subtitle1};
+`
 
-  const { id } = router.query;
-  if (typeof id !== 'string') return null;
+const EventsHostedByClub = ({ events }: { events: Array<BasicEvent> }) => {
+  const upcomingEvents = events.filter((e) => e.startTime.isAfter(moment()))
 
-  const { name, description } = clubs.get(parseInt(id));
+  const pastEvents = events.filter((e) => e.startTime.isBefore(moment()))
 
   return (
     <div>
-      <h1>Club Details</h1>
-      <p>{id}</p>
-      <p>{name}</p>
-      <p>{description}</p>
+      <EventsTitle>Upcoming Events</EventsTitle>
+      {upcomingEvents.map((e) => {
+        return <p>h</p>
+      })}
+      <PastEventsTitle>Past Events</PastEventsTitle>
+      {pastEvents.map((e) => {
+        return (
+          <EventCard>
+            <EventCardImg src={e.backgroundImageURL} alt="" />
+            <EventCardDetails>
+              <EventCardTitle>{e.name}</EventCardTitle>
+              <EventCardDate>
+                {device.mobileS ? (
+                  <p>
+                    {e.startTime.format('dddd')}, {e.startTime.format('LL')}
+                    <br />
+                    {e.startTime.format('LT')} - {e.endTime.format('LT')} ET
+                  </p>
+                ) : (
+                  <p>
+                    {e.startTime.format('dddd')}, {e.startTime.format('LL')} |{' '}
+                    {e.startTime.format('LT')} - {e.endTime.format('LT')} ET
+                  </p>
+                )}
+              </EventCardDate>
+              <div style={{ display: 'flex', justifyContent: 'flex-end' }}>
+                {e.tags.map((t) => {
```

Comment:
Again, for consistency it would be nice to render sub-components from another function cause we do that in other spots and it helps maintainability.